no-fail-fast support for tool testsuites

jchecahi opened the pull request:
~~This commit adds a change to pass "--no-fail-fast" flag to `cargo test` inside `tool::prepare_tool_cargo()` so there is no need to do it manually in each `Step` trait implementation in src/bootstrap/test.rs.~~

~~Also, removes the flag from test.rs where prepare_tool_cargo() is called so cargo doesn't complain because the flag has been passed twice.~~

This commit adds `--no-fail-fast` flag to each `cargo test`
command in each tool Step trait implementation (`miri`, `rustfmt` and `clippy`).


Fixes #108261

jchecahi commented:
I have tested this by enforcing some tests to fail by adding an `assert!(1==2);` statement in `clippy`, `rustfmt` and `miri` tests. Without the commit, the testsuite stops running after a failure, with the commit, it keeps going until the end.

ozkanonur commented:
Changing behaviour of test suite in tool module doesn't seem right. It also duplicates the control testing to multiple modules which isn't really good. You can pass `--no-fail-fast` for Miri, Rustfmt and Clippy same as in the other `Step` implementations in test module.

ozkanonur commented:
@rustbot author

Commit:
Add no-fail-fast support to miri, rustfmt and clippy testsuites

This commit adds `--no-fail-fast` flag to each `cargo test`
command in each tool Step trait implementation.

Fixes #108261

jchecahi commented:
@ozkanonur thanks for the input, I've made the changes you recommended. Let me make sure I understood the reasoning behind your advice. IIUIC the changes related to the testsuites should be placed always in `src/bootstrap/test.rs`. Following that reasoning, if the goal is to gather this kind of duplicated logic (as suggested in #104198), the proper way to do it would be to create a function in `src/bootstrap/test.rs` that takes care of that. Did I understand it correctly?

ozkanonur commented:
We already pass the test flags in test module, if we do it in other modules as well, it would be not good for us to work/debug on it.

And yes, creating `fn` could be quite nice.

ozkanonur commented:
This looks good for now, we can do the code unification in seperated PR pointing the related issue. Thank you for the contribution.

@bors +r rollup

ozkanonur commented:
@bors +r rollup

ozkanonur commented:
@bors r+ rollup

bors commented:
:pushpin: Commit 2f163554866e1b4b348811e9df6a4a753beb0abf has been approved by `ozkanonur`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=ozkanonur 2f163554866e1b4b348811e9df6a4a753beb0abf -->
<!-- homu: {"type":"Approved","sha":"2f163554866e1b4b348811e9df6a4a753beb0abf","approver":"ozkanonur","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->
